Frontend/src/hooks/useLoadAllCollection.ts

```diff
@@ -0,0 +1,64 @@
+import axios from "axios"
+import React, { useContext } from "react"
+import StateContext from "../state/stateContext"
+
+interface Props {
+  collections: any[]
+  setCollections: any
+  isCollectionsLoading: boolean
+  setIsCollectionsLoading: any
+}
+
+export default function useLoadAllCollections({
+  collections,
+  setCollections,
+  isCollectionsLoading,
+  setIsCollectionsLoading,
+}: Props) {
+  const { daoContract } = useContext(StateContext)
+
+  React.useEffect(() => {
+    //connect to Ballot smart contract
+
+    //get collections
+    const getCollections = async () => {
+      if (!daoContract || collections.length > 0 || isCollectionsLoading) return
+
+      //get CID of items of each collection
+      //get current amount of votes for each collection
+      //get issuer of each collection
+      try {
+        setIsCollectionsLoading(true)
+
+        const collections = []
+        const proposals = await daoContract.getAllProposals()
```

Comment:
Perhaps it makes sense to add here a `setProposals(proposals)`? And then in `Vote.tsx` we can take them from state

Reply:
Well, no. Actually makes more sense to add another hook for proposals

Reply:
Yeah I also thought of this. The reason I didn't do it is that we will have old state at some point if we only fetch the proposals once for the entire app lifetime.
If in the meantime there are new proposals (which we are planning to do) we don't fetch it again. So would rather fetch proposals on every page change.

Reply:
OK :+1: